pkg/controller/daemon/daemon_controller.go

```diff
@@ -774,9 +774,14 @@ func (dsc *DaemonSetsController) getNodesToDaemonPods(ds *apps.DaemonSet) (map[s
 	// Group Pods by Node name.
 	nodeToDaemonPods := make(map[string][]*v1.Pod)
 	for _, pod := range claimedPods {
-		nodeName := pod.Spec.NodeName
+		nodeName, err := util.GetTargetNodeName(pod)
+		if err != nil {
+			return nil, err
+		}
+
 		nodeToDaemonPods[nodeName] = append(nodeToDaemonPods[nodeName], pod)
 	}
+
```

Comment:
Because `nodeToDaemonPods` includes pods that's not scheduled yet (no nodeName set), in `podsShouldBeOnNode`, the pods without nodeName set should be killed first when excess daemon pods are found.

Reply:
done, sort the pods by phase & creation timestamp.

Reply:
Just revert it; the node without `nodeName` maybe the new one we want to create.

Reply:
> Just revert it; the node without `nodeName` maybe the new one we want to create.

I'm not sure if I understand why this is reverted. If a DaemonSet pod is already scheduled on this node, shouldn't other pending pods be deleted first?